internal/liqonet/tunnelEndpointCreator/tunnelEndpointCreator-config.go

```diff
@@ -71,30 +81,91 @@ func (tec *TunnelEndpointCreator) UpdateConfiguration(reservedSubnets []string)
 			//free subnet in ipam
 			klog.Infof("Freeing reserved subnet %s", subnet)
 			if err := tec.IPManager.FreeReservedSubnet(subnet); err != nil {
-				return err
+				klog.Error(err)
 			}
 			//remove the subnet from the reserved ones
-			tec.removeSubnetFromConfig(subnet)
+			tec.removeReservedSubnet(subnet)
 		}
 	}
 	if len(addedSubnets) > 0 {
 		for _, subnet := range addedSubnets {
 			klog.Infof("Reserving subnet %s", subnet)
 			if err := tec.IPManager.AcquireReservedSubnet(subnet); err != nil {
+				klog.Error(err)
+			}
+			tec.addReservedSubnet(subnet)
+		}
+	}
+	return nil
+}
+
+func (tec *TunnelEndpointCreator) UpdatePools(additionalPools []string) error {
```

Comment:
this function does the same as the UpdateReservedSubnets. could we move the logic outside these functions, having it in one place. We call it to check if the configuration in the clusterConfig resource has changed and then return what subnets that have been added or removed. Then the UpdateReservedSubnets and UpdatePools update accordingly the configuration inside the `TunnelEndpointCreator` instance.

Reply:
Good tip